kroxylicious-additional-filters/kroxylicious-multitenant/src/test/java/io/kroxylicious/proxy/filter/multitenant/MultiTenantTransformationFilterTest.java

```diff
@@ -98,18 +120,26 @@ public static Stream<Arguments> responses() throws Exception {
 
     @ParameterizedTest(name = "{0}")
     @MethodSource(value = "responses")
-    void responseTransformed(@SuppressWarnings("unused") String testName, ApiMessageType apiMessageType, RequestHeaderData header, ApiMessageTestDef responseTestDef) {
+    void responseTransformed(@SuppressWarnings("unused") String testName, ApiMessageType apiMessageType, RequestHeaderData header, ApiMessageTestDef responseTestDef)
+            throws Exception {
         var response = responseTestDef.message();
         // marshalled the response object back to json, this is used for comparison later.
         var responseWriter = responseConverterFor(apiMessageType).writer();
 
         var marshalled = responseWriter.apply(response, header.requestApiVersion());
 
+        when(context.sniHostname()).thenReturn(TENANT_1);
```

Comment:
Not something to tackle in this PR but this does make me wonder if we should be providing some sort of stub context rather than mocking it.

Reply:
Yes, I had the same thought.  We need to provide a good stub that allows tenant authors to unit test their filters.  Agree - one for later.